File: food-delivery/app/Models/Admin/ProdukModel.php
<?php

namespace App\Models\Admin;

use CodeIgniter\Model;

class ProdukModel extends Model
{
    protected $table = 'produk';
    protected $useTimestamps = true;
    protected $allowedFields = ['nama_produk', 'harga', 'deskripsi', 'gambar', 'status', 'kategori'];

    public function getRulesValidation($method)
    {
        if ($method == 'update') {
            $imgRules = 'max_size[gambar,1024]|is_image[gambar]|ext_in[gambar,png,jpg,jpeg]';
        } else {
            $imgRules = 'uploaded[gambar]|max_size[gambar,1024]|is_image[gambar]|ext_in[gambar,png,jpg,jpeg]';
        }

        $rulesValidation = [
            'nama_produk' => [
                'label' => 'nama produk',
                'rules' => 'required',
                'errors' => [
                    'required' => '{field} harus diisi.'
                ]
            ],
            'harga' => [
                'label' => 'harga',
                'rules' => 'required',
                'errors' => [
                    'required' => '{field} harus diisi.'
                ]
            ],
            'deskripsi' => [
                'label' => 'deskripsi',
                'rules' => 'required',
                'errors' => [
                    'required' => '{field} harus diisi.'
                ]
            ],
            'kategori' => [
                'label' => 'kategori',
                'rules' => 'required',
                'errors' => [
                    'required' => '{field} harus diisi.'
                ]
            ],
            'gambar' => [
                'label' => 'gambar',
                'rules' => $imgRules,
                'errors' => [
                    'uploaded'  => '{field} harus dipilih.',
                    'max_size'  => '{field} melebihi ukuran yang ditentukan (max. 1MB).',
                    'is_image'  => 'format {field} tidak sesuai.',
                    'ext_in'    => 'format {field} tidak sesuai.'
                ]
            ],
        ];

        return $rulesValidation;
    }

    public function ajaxGetData($length, $start)
    {
        $result = $this->orderBy('id', 'desc')
            ->findAll($length, $start);

        return $result;
    }

    public function ajaxGetDataSearch($search, $length, $start)
    {
        $result = $this->like('nama_produk', $search)
            ->orLike('harga', $search)
            ->orLike('deskripsi', $search)
            ->orLike('kategori', $search)
            ->orderBy('id', 'desc')
            ->findAll($length, $start);

        return $result;
    }

    public function ajaxGetTotal()
    {
        $result = $this->countAll();

        if (isset($result)) {
            return $result;
        }

        return 0;
    }

    public function ajaxGetTotalSearch($search)
    {
        $result = $this->like('nama_produk', $search)
            ->orLike('harga', $search)
            ->orLike('deskripsi', $search)
            ->orLike('kategori', $search)
            ->countAllResult();

        return $result;
    }
}

File: food_delivery/lib/src/model/ProdukModel.dart
import 'dart:convert';

List<ProdukModel> produkFromJson(String str) {
  final jsonData = jsonDecode(str);
  return List<ProdukModel>.from(jsonData.map((x) => ProdukModel.fromJson(x)));
}

class ProdukModel {
  String id;
  String nama_produk;
  String harga;
  String deskripsi;
  String gambar;
  String status;
  String kategori;

  ProdukModel({
    this.id,
    this.nama_produk,
    this.harga,
    this.deskripsi,
    this.gambar,
    this.status,
    this.kategori,
  });

  factory ProdukModel.fromJson(Map<String, dynamic> json) => ProdukModel(
        id: json['id'],
        nama_produk: json['nama_produk'],
        harga: json['harga'],
        deskripsi: json['deskripsi'],
        gambar: json['gambar'],
        status: json['status'],
        kategori: json['kategori'],
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'nama_produk': nama_produk,
        'harga': harga,
        'deskripsi': deskripsi,
        'gambar': gambar,
        'status': status,
        'kategori': kategori
      };
}
